formatter/src/main/java/org/togetherjava/tjbot/formatter/tokenizer/TokenType.java

```diff
@@ -1,137 +1,273 @@
+    // Their patterns are applied from top to bottom. So entries that are not prefix-free
+    // like "--" and "-" must be ordered with "--" before "-" to correctly match.
+
+    // Keywords
+    CLASS("class", Attribute.KEYWORD),
+    ENUM("enum", Attribute.KEYWORD),
+    RECORD("record", Attribute.KEYWORD),
+    INTERFACE("interface", Attribute.KEYWORD),
+    IMPORT("import", Attribute.KEYWORD),
+    NULL("null", Attribute.KEYWORD),
+    EXTENDS("extends", Attribute.KEYWORD),
+    IMPLEMENTS("implements", Attribute.KEYWORD),
+    NEW("new", Attribute.KEYWORD),
+    RETURN("return", Attribute.KEYWORD),
+    PACKAGE("package", Attribute.KEYWORD),
+    THIS("this", Attribute.KEYWORD),
+    YIELD("yield", Attribute.KEYWORD),
+    SUPER("super", Attribute.KEYWORD),
+    ASSERT("assert", Attribute.KEYWORD),
+    CONST("const", Attribute.KEYWORD),
+    DEFAULT("default", Attribute.KEYWORD),
+    FINALLY("finally", Attribute.KEYWORD),
+    THROWS("throws", Attribute.KEYWORD),
+    THROW("throw", Attribute.KEYWORD),
+
+    // Access modifiers
+    PUBLIC("public", Attribute.KEYWORD),
+    PRIVATE("private", Attribute.KEYWORD),
+    PROTECTED("protected", Attribute.KEYWORD),
+    STATIC("static", Attribute.KEYWORD),
+    SEALED("sealed", Attribute.KEYWORD),
+    NON_SEALED("non-sealed", Attribute.KEYWORD),
+    FINAL("final", Attribute.KEYWORD),
+    ABSTRACT("abstract", Attribute.KEYWORD),
+    NATIVE("native", Attribute.KEYWORD),
+    STRICTFP("strictfp", Attribute.KEYWORD),
+    SYNCHRONIZED("synchronized", Attribute.KEYWORD),
+    TRANSIENT("transient", Attribute.KEYWORD),
+    VOLATILE("volatile", Attribute.KEYWORD),
+
+    // Primitives
+    VOID("void", Attribute.KEYWORD),
+    INT("int", Attribute.KEYWORD),
+    LONG("long", Attribute.KEYWORD),
+    SHORT("short", Attribute.KEYWORD),
+    BYTE("byte", Attribute.KEYWORD),
+    BOOLEAN("boolean", Attribute.KEYWORD),
+    FLOAT("float", Attribute.KEYWORD),
+    DOUBLE("double", Attribute.KEYWORD),
+    CHAR("char", Attribute.KEYWORD),
+
+    // Flow control
+    IF("if", Attribute.KEYWORD),
+    // Technically two keywords, but we want it to stick together when formatting
+    ELSE_IF("else if", Attribute.KEYWORD),
+    ELSE("else", Attribute.KEYWORD),
+    FOR("for", Attribute.KEYWORD),
+    WHILE("while", Attribute.KEYWORD),
+    DO("do", Attribute.KEYWORD),
+    BREAK("break", Attribute.KEYWORD),
+    CONTINUE("continue", Attribute.KEYWORD),
+    SWITCH("switch", Attribute.KEYWORD),
+    CASE("case", Attribute.KEYWORD),
+    TRY("try", Attribute.KEYWORD),
+    CATCH("catch", Attribute.KEYWORD),
+    GOTO("goto", Attribute.KEYWORD),
+
+    // Braces
     OPEN_PARENTHESIS("("),
     CLOSE_PARENTHESIS(")"),
     OPEN_BRACES("{"),
     CLOSE_BRACES("}"),
     OPEN_BRACKETS("["),
     CLOSE_BRACKETS("]"),
+
+    // General
+    DOT("."),
     SEMICOLON(";"),
     METHOD_REFERENCE("::"),
-    COLON(":", false, true), // technically not a "real" operator but used in an enhanced for loop
```

Comment:
Not needed anymore?

Reply:
it is needed, so that the lexxer recognizes it as individual token.

its still part of the list somewhere, i reordered a lot of stuff

Reply:
found it :)

![colon](https://i.imgur.com/O4evx4s.png)